### PR TITLE
Refresh FileList on current folder update

### LINKED ISSUE #845: Refresh current folder view
What would be the way to refresh the current directory in the files app?

In my app I dont know if/what files have changed so I just need to do a full refresh.

Previously I would use this 

```js
const fileList = OCA.Files?.App?.fileList ?? OCA?.Sharing?.PublicApp?.fileList
if (!fileList) {
  return
}
fileList.changeDirectory(fileList.getCurrentDirectory(), true, true)
```
https://github.com/Raudius/files_scripts/blob/v3.0.0/src/files.ts#L3

### PR DESCRIPTION
**Pull request by Raudius**
* Resolves: https://github.com/nextcloud-libraries/nextcloud-files/issues/845

## Summary
This PR subscribes the `FileList` view to the file-updated event. If the file updated matches the current folder it re-fetches the current view.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files/src/views/FilesList.vue [hunk 1]
```diff
@@ -556,6 +557,17 @@ export default defineComponent({
 			showError(this.t('files', 'Unknown error during upload'))
 		},
 
+		/**
+		 * Refreshes the current folder on update.
+		 *
+		 * @param {Node} node is the file/folder being updated.
+ 		 */
+		onUpdatedNode(node) {
+			if (node.fileid === this.currentFolder.fileid) {
```
Comment (skjnldsv) [inline apps/files/src/views/FilesList.vue:566]: To be safe?

```suggestion
			if (node?.fileid === this.currentFolder?.fileid) {
```

**Review by skjnldsv — APPROVED** [review]

**Comment by hschletz** [comment]
This PR won't work because the function names mismatch: subscribe() uses "onUpdateNode", but actual name is "onUpdatedNode".

**Comment by Raudius** [comment]
Nice catch @hschletz! I have now fixed it and tested again using my app which already integrates the changes:

https://github.com/Raudius/files_scripts/blob/master/src/files.ts#L10

File: apps/files/src/views/FilesList.vue [hunk 2]
```diff
@@ -436,6 +436,7 @@ export default defineComponent({
 
 	mounted() {
 		this.fetchContent()
+		subscribe('files:node:updated', this.onUpdatedNode)
```
Comment (artonge) [inline apps/files/src/views/FilesList.vue:439]: No need to unsubscribe?

Comment (artonge) [reply apps/files/src/views/FilesList.vue:439]: If not, then a comment would be appreciated :)

Comment (Raudius) [reply apps/files/src/views/FilesList.vue:439]: Thanks for the tip, I am not familiar with the event bus.

Should be fixed now

**Review by artonge — APPROVED** [review]

**Comment by artonge** [comment]
@Raudius please rebase and compile again, then it will be good to merge :)

**Pull request merged**